## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :x: [more...](#290c2bfb-97be-460c-b41c-4861345ecafa) |
| mint-zoned.sh | :x: [more...](#62abab84-3f4c-42b5-b104-5ec323eef7f6) |
| mint-gateway-nas.sh | :x: [more...](#9decde10-7754-4f3b-9c33-f56e3d594e5e) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#a784d998-57ca-4ce9-ae8f-6a92f41470dc) |
<a name="a784d998-57ca-4ce9-ae8f-6a92f41470dc"></a>
### 12463-a2e81ae/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:31078
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a4663eab43a2:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-tvgwms2wnzthgo",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364991,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 1686B79C88A2FD6B, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-tvgwms2wnzthgo",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364991,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 1686B79C88A2FD6B, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-php tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-tvgwms2wnzthgo",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364991,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 1686B79C88A2FD6B, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-ruby tests ... FAILED in 3 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2151.16,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(1/15) Running awscli tests ... FAILED in 13 seconds
{
  "name": "awscli",
  "duration": 11384,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(1/15) Running healthcheck tests ... done in 0 seconds
(2/15) Running mc tests ... FAILED in 0 seconds
(2/15) Running minio-dotnet tests ... done in 3 seconds
(3/15) Running minio-go tests ... FAILED in 4 seconds
{
  "args": {
    "bucketName": "minio-go-test-i4qztlu1blozeho3",
    "region": "eu-west-1"
  },
  "duration": 3564,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(3/15) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 318,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = XMinioServerNotInitialized, message = Server not initialized, please try again., bucketName = minio-java-test-iaf4a6, objectName = null, resource = /minio-java-test-iaf4a6, requestId = 1686B7A2D854A540, hostId = null)\nrequest={method=PUT, url=http://minio-dev8.minio.io:31078/minio-java-test-iaf4a6, headers=Host: minio-dev8.minio.io:31078\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210608T211025Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210608/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 300\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1686B7A2D854A540\nX-Xss-Protection: 1; mode=block\nDate: Tue, 08 Jun 2021 21:10:25 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.makeBucket(MinioClient.java:2276), FunctionalTest.testBucketApi(FunctionalTest.java:446), FunctionalTest.testBucketApiCases(FunctionalTest.java:471), FunctionalTest.makeBucket(FunctionalTest.java:502), FunctionalTest.runBucketTests(FunctionalTest.java:3708), FunctionalTest.runTests(FunctionalTest.java:3782), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(3/15) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 9,
  "status": "FAIL",
  "error": "S3Error: Server not initialized, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:79:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:684:10) at endReadableNT (internal/streams/readable.js:1327:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(3/15) Running minio-py tests ... FAILED in 10 minutes and 0 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-4881cdc3-7c66-48b6-beca-bfdd6d855418",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-dev8.minio.io', port=31078): Max retries exceeded with url: /minio-py-test-4881cdc3-7c66-48b6-beca-bfdd6d855418 (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 169, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev8.minio.io', port=31078): Max retries exceeded with url: /minio-py-test-4881cdc3-7c66-48b6-beca-bfdd6d855418 (Caused by ResponseError('too many 503 error responses',))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 600338
}
(3/15) Running s3cmd tests ... FAILED in 46 seconds
{
  "name": "s3cmd",
  "duration": "45426",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(3/15) Running s3select tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-d9b7da33-caaf-4209-ae8f-5df1a159d0c8"
  },
  "duration": 600559,
  "message": "HTTPConnectionPool(host='minio-dev8.minio.io', port=31078): Max retries exceeded with url: /s3select-test-d9b7da33-caaf-4209-ae8f-5df1a159d0c8 (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev8.minio.io', port=31078): Max retries exceeded with url: /s3select-test-d9b7da33-caaf-4209-ae8f-5df1a159d0c8 (Caused by ResponseError('too many 503 error responses',))\n",
  "status": "FAIL"
}
(3/15) Running security tests ... done in 0 seconds

Executed 3 out of 15 tests successfully.
```
<a name="9decde10-7754-4f3b-9c33-f56e3d594e5e"></a>
### 12463-a2e81ae/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:31406
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e0dbd2d4a6ac:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-rze3tc16tr2yjr",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 317,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-dev8.minio.io:31406/aws-sdk-go-test-rze3tc16tr2yjr\": dial tcp 72.28.97.61:31406: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-rze3tc16tr2yjr",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 317,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-dev8.minio.io:31406/aws-sdk-go-test-rze3tc16tr2yjr\": dial tcp 72.28.97.61:31406: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-php tests ... FAILED in 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-rze3tc16tr2yjr",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 317,
  "error": "RequestError: send request failed\ncaused by: Put \"http://minio-dev8.minio.io:31406/aws-sdk-go-test-rze3tc16tr2yjr\": dial tcp 72.28.97.61:31406: connect: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-ruby tests ... FAILED in 4 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 3155.18,
  "error": "Failed to open TCP connection to minio-dev8.minio.io:31406 (Connection refused - connect(2) for \"minio-dev8.minio.io\" port 31406)",
  "status": "FAIL"
}
(1/15) Running awscli tests ... FAILED in 9 seconds
{"name": "awscli", "duration": 6128, "function": "make_bucket\n", "status": "FAIL", "error": "Could not connect to the endpoint URL: "http://minio-dev8.minio.io:31406/awscli-mint-test-bucket-16663""}
(1/15) Running healthcheck tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 15,
  "error": "Get \"http://minio-dev8.minio.io:31406/minio/health/live\": dial tcp 72.28.97.61:31406: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/15) Running mc tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 15,
  "error": "Get \"http://minio-dev8.minio.io:31406/minio/health/live\": dial tcp 72.28.97.61:31406: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/15) Running minio-dotnet tests ... done in 45 seconds
(2/15) Running minio-go tests ... FAILED in 10 seconds
{
  "args": {
    "bucketName": "minio-go-test-vhuxi4sftjiaxy0y",
    "region": "eu-west-1"
  },
  "duration": 10249,
  "error": "Put \"http://minio-dev8.minio.io:31406/minio-go-test-vhuxi4sftjiaxy0y/\": dial tcp 72.28.97.61:31406: connect: connection refused",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(2/15) Running minio-java tests ... FAILED in 0 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 66,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to minio-dev8.minio.io/72.28.97.61:31406 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.MinioClient.execute(MinioClient.java:647), io.minio.MinioClient.makeBucket(MinioClient.java:2276), FunctionalTest.testBucketApi(FunctionalTest.java:446), FunctionalTest.testBucketApiCases(FunctionalTest.java:471), FunctionalTest.makeBucket(FunctionalTest.java:502), FunctionalTest.runBucketTests(FunctionalTest.java:3708), FunctionalTest.runTests(FunctionalTest.java:3782), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(2/15) Running minio-js tests ... FAILED in 3 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 1823,
  "status": "FAIL",
  "error": "Error: connect ECONNREFUSED 72.28.97.61:31406 at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16)"
}
(2/15) Running minio-py tests ... FAILED in 8 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-8df77f52-6cb1-45bb-b99c-406e323cea2b",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-dev8.minio.io', port=31406): Max retries exceeded with url: /minio-py-test-8df77f52-6cb1-45bb-b99c-406e323cea2b (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f6e593ba518>: Failed to establish a new connection: [Errno 111] Connection refused',))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 170, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 96, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 86, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 706, in urlopen\n    chunked=chunked,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 394, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 234, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.6/http/client.py\", line 1281, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1327, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1276, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1042, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.6/http/client.py\", line 980, in send\n    self.connect()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 200, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 182, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f6e593ba518>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 169, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 756, in urlopen\n    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev8.minio.io', port=31406): Max retries exceeded with url: /minio-py-test-8df77f52-6cb1-45bb-b99c-406e323cea2b (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f6e593ba518>: Failed to establish a new connection: [Errno 111] Connection refused',))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 8102
}
(2/15) Running s3cmd tests ... FAILED in 1 seconds
{
  "name": "s3cmd",
  "duration": "305",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-17727/s3cfg mb s3://s3cmd-test-bucket-10833\nProblem: <class 'ConnectionRefusedError: [Errno 111] Connection refused\nS3cmd:   2.1.0\npython:   3.6.9 (default, Oct  8 2020, 12:12:24) \n[GCC 8.4.0]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3121, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3030, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 251, in cmd_bucket_create\n    response = s3.bucket_create(uri.bucket(), cfg.bucket_location)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 408, in bucket_create\n    response = self.send_request(request)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 1281, in send_request\n    conn = ConnMan.get(self.get_hostname(resource['bucket']))\n  File \"/usr/local/lib/python3.6/dist-packages/S3/ConnMan.py\", line 251, in get\n    conn.c.connect()\n  File \"/usr/lib/python3.6/http/client.py\", line 952, in connect\n    (self.host,self.port), self.timeout, self.source_address)\n  File \"/usr/lib/python3.6/socket.py\", line 724, in create_connection\n    raise err\n  File \"/usr/lib/python3.6/socket.py\", line 713, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}
(2/15) Running s3select tests ... FAILED in 6 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-65fb63f8-9bb0-4f33-9ca9-81250fb6abb6"
  },
  "duration": 6105,
  "message": "HTTPConnectionPool(host='minio-dev8.minio.io', port=31406): Max retries exceeded with url: /s3select-test-65fb63f8-9bb0-4f33-9ca9-81250fb6abb6 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fbc0021ef60>: Failed to establish a new connection: [Errno 111] Connection refused',))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 170, in _new_conn\n    (self._dns_host, self.port), self.timeout, **extra_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 96, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/connection.py\", line 86, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 706, in urlopen\n    chunked=chunked,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 394, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 234, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.6/http/client.py\", line 1281, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1327, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1276, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.6/http/client.py\", line 1042, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.6/http/client.py\", line 980, in send\n    self.connect()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 200, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connection.py\", line 182, in _new_conn\n    self, \"Failed to establish a new connection: %s\" % e\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fbc0021ef60>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 756, in urlopen\n    method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev8.minio.io', port=31406): Max retries exceeded with url: /s3select-test-65fb63f8-9bb0-4f33-9ca9-81250fb6abb6 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fbc0021ef60>: Failed to establish a new connection: [Errno 111] Connection refused',))\n",
  "status": "FAIL"
}
(2/15) Running security tests ... done in 0 seconds

Executed 2 out of 15 tests successfully.
```
<a name="62abab84-3f4c-42b5-b104-5ec323eef7f6"></a>
### 12463-a2e81ae/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:30465
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 68ecfd73e3d3:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... FAILED in 2 minutes and 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-gidawxkepa5sh5",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 123291,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 1686B7644DDB8C9B, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-java tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-gidawxkepa5sh5",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 123291,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 1686B7644DDB8C9B, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-php tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-gidawxkepa5sh5",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 123291,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 1686B7644DDB8C9B, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/15) Running aws-sdk-ruby tests ... FAILED in 3 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2166.49,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(1/15) Running awscli tests ... FAILED in 11 seconds
{"name": "awscli", "duration": 9676, "function": "make_bucket\n", "status": "FAIL", "error": "Could not connect to the endpoint URL: "http://minio-c3.minio.io:30465/awscli-mint-test-bucket-28399""}
(1/15) Running healthcheck tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": null,
  "duration": 8,
  "error": "Get \"http://minio-c3.minio.io:30465/minio/health/live\": dial tcp 72.28.97.59:30465: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/15) Running mc tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": null,
  "duration": 8,
  "error": "Get \"http://minio-c3.minio.io:30465/minio/health/live\": dial tcp 72.28.97.59:30465: connect: connection refused",
  "function": "testLivenessEndpoint",
  "message": "GET request failed",
  "name": "healthcheck",
  "status": "FAIL"
}
(1/15) Running minio-dotnet tests ... done in 3 seconds
(2/15) Running minio-go tests ... FAILED in 4 seconds
{
  "args": {
    "bucketName": "minio-go-test-wxb0ywc4stzobpud",
    "region": "eu-west-1"
  },
  "duration": 4561,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testGetObjectClosedTwiceV2",
  "status": "FAIL"
}
(2/15) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 96,
  "status": "FAIL",
  "error": "java.net.ConnectException: Failed to connect to minio-c3.minio.io/72.28.97.59:30465 >>> [okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.kt:297), okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:207), okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:226), okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:106), okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:74), okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:255), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.MinioClient.execute(MinioClient.java:647), io.minio.MinioClient.makeBucket(MinioClient.java:2276), FunctionalTest.testBucketApi(FunctionalTest.java:446), FunctionalTest.testBucketApiCases(FunctionalTest.java:471), FunctionalTest.makeBucket(FunctionalTest.java:502), FunctionalTest.runBucketTests(FunctionalTest.java:3708), FunctionalTest.runTests(FunctionalTest.java:3782), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(2/15) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 15,
  "status": "FAIL",
  "error": "Error: connect ECONNREFUSED 72.28.97.59:30465 at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1146:16)"
}
(2/15) Running minio-py tests ... FAILED in 6 minutes and 1 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-481df919-c0f4-4bc2-9b12-34e51aa8a91c",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='minio-c3.minio.io', port=30465): Max retries exceeded with url: /minio-py-test-481df919-c0f4-4bc2-9b12-34e51aa8a91c (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 169, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-c3.minio.io', port=30465): Max retries exceeded with url: /minio-py-test-481df919-c0f4-4bc2-9b12-34e51aa8a91c (Caused by ResponseError('too many 503 error responses',))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 360681
}
(2/15) Running s3cmd tests ... FAILED in 1 seconds
{
  "name": "s3cmd",
  "duration": "294",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-7702/s3cfg mb s3://s3cmd-test-bucket-1807\nProblem: <class 'ConnectionRefusedError: [Errno 111] Connection refused\nS3cmd:   2.1.0\npython:   3.6.9 (default, Oct  8 2020, 12:12:24) \n[GCC 8.4.0]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3121, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3030, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 251, in cmd_bucket_create\n    response = s3.bucket_create(uri.bucket(), cfg.bucket_location)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 408, in bucket_create\n    response = self.send_request(request)\n  File \"/usr/local/lib/python3.6/dist-packages/S3/S3.py\", line 1281, in send_request\n    conn = ConnMan.get(self.get_hostname(resource['bucket']))\n  File \"/usr/local/lib/python3.6/dist-packages/S3/ConnMan.py\", line 251, in get\n    conn.c.connect()\n  File \"/usr/lib/python3.6/http/client.py\", line 952, in connect\n    (self.host,self.port), self.timeout, self.source_address)\n  File \"/usr/lib/python3.6/socket.py\", line 724, in create_connection\n    raise err\n  File \"/usr/lib/python3.6/socket.py\", line 713, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}
(2/15) Running s3select tests ... FAILED in 8 minutes and 0 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-1fce123e-e5d8-4718-9623-50653c3b6e69"
  },
  "duration": 480338,
  "message": "HTTPConnectionPool(host='minio-c3.minio.io', port=30465): Max retries exceeded with url: /s3select-test-1fce123e-e5d8-4718-9623-50653c3b6e69 (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 624, in make_bucket\n    headers=headers,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 796, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 1 more time]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-c3.minio.io', port=30465): Max retries exceeded with url: /s3select-test-1fce123e-e5d8-4718-9623-50653c3b6e69 (Caused by ResponseError('too many 503 error responses',))\n",
  "status": "FAIL"
}
(2/15) Running security tests ... done in 0 seconds

Executed 2 out of 15 tests successfully.
```
<a name="290c2bfb-97be-460c-b41c-4861345ecafa"></a>
### 12463-a2e81ae/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:31647
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 113bb8db4c82:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 1 minutes and 4 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 30 seconds
(8/15) Running minio-dotnet tests ... done in 29 seconds
(9/15) Running minio-go tests ... done in 48 seconds
(10/15) Running minio-java tests ... FAILED in 18 seconds
{
  "name": "minio-java",
  "function": "getObject()",
  "args": "[multi-part upload, offset, length]",
  "duration": 199,
  "status": "FAIL",
  "error": "java.net.SocketException: Connection reset by peer (Write failed) >>> [java.base/java.net.SocketOutputStream.socketWrite0(Native Method), java.base/java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:110), java.base/java.net.SocketOutputStream.write(SocketOutputStream.java:150), okio.OutputStreamSink.write(JvmOkio.kt:53), okio.AsyncTimeout$sink$1.write(AsyncTimeout.kt:103), okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.kt:235), okio.RealBufferedSink.write(RealBufferedSink.kt:145), okhttp3.internal.http1.Http1ExchangeCodec$KnownLengthSink.write(Http1ExchangeCodec.kt:271), okio.ForwardingSink.write(ForwardingSink.kt:29), okhttp3.internal.connection.Exchange$RequestBodySink.write(Exchange.kt:218), okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.kt:235), okio.RealBufferedSink.write(RealBufferedSink.kt:193), io.minio.HttpRequestBody.writeTo(HttpRequestBody.java:78), okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.kt:59), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:34), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:83), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:76), okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109), okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:201), okhttp3.internal.connection.RealCall.execute(RealCall.kt:154), io.minio.MinioClient.execute(MinioClient.java:647), io.minio.MinioClient.uploadPart(MinioClient.java:4747), io.minio.MinioClient.putObject(MinioClient.java:2738), io.minio.MinioClient.putObject(MinioClient.java:2830), FunctionalTest.testGetObject(FunctionalTest.java:1136), FunctionalTest.getObject(FunctionalTest.java:1229), FunctionalTest.runObjectTests(FunctionalTest.java:3752), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(10/15) Running minio-js tests ... done in 40 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 24 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 3 seconds
(13/15) Running s3select tests ... done in 4 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/common-main.go
+++ b/cmd/common-main.go
@@ -22,7 +22,6 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"encoding/gob"
-	"encoding/hex"
 	"errors"
 	"fmt"
 	"math/rand"
@@ -339,27 +338,10 @@ func handleCommonEnvVars() {
 	switch {
 	case env.IsSet(config.EnvKMSSecretKey) && env.IsSet(config.EnvKESEndpoint):
 		logger.Fatal(errors.New("ambigious KMS configuration"), fmt.Sprintf("The environment contains %q as well as %q", config.EnvKMSSecretKey, config.EnvKESEndpoint))
-	case env.IsSet(config.EnvKMSMasterKey) && env.IsSet(config.EnvKESEndpoint):
-		logger.Fatal(errors.New("ambigious KMS configuration"), fmt.Sprintf("The environment contains %q as well as %q", config.EnvKMSMasterKey, config.EnvKESEndpoint))
 	}
 
 	if env.IsSet(config.EnvKMSSecretKey) {
 		GlobalKMS, err = kms.Parse(env.Get(config.EnvKMSSecretKey, ""))
-		if err != nil {
-			logger.Fatal(err, "Unable to parse the KMS secret key inherited from the shell environment")
-		}
-	} else if env.IsSet(config.EnvKMSMasterKey) {
-		// FIXME: remove this block by June 2021
-		logger.LogIf(GlobalContext, fmt.Errorf("legacy KMS configuration, this environment variable %q is deprecated and will be removed by June 2021", config.EnvKMSMasterKey))
-		v := strings.SplitN(env.Get(config.EnvKMSMasterKey, ""), ":", 2)
-		if len(v) != 2 {
-			logger.Fatal(errors.New("invalid "+config.EnvKMSMasterKey), "Unable to parse the KMS secret key inherited from the shell environment")
-		}
-		secretKey, err := hex.DecodeString(v[1])
-		if err != nil {
-			logger.Fatal(err, "Unable to parse the KMS secret key inherited from the shell environment")
-		}
-		GlobalKMS, err = kms.New(v[0], secretKey)
 		if err != nil {
 			logger.Fatal(err, "Unable to parse the KMS secret key inherited from the shell environment")
 		}

--- a/dockerscripts/docker-entrypoint.sh
+++ b/dockerscripts/docker-entrypoint.sh
@@ -71,17 +71,6 @@ docker_kms_secret_encryption_env() {
     fi
 }
 
-## Legacy
-## Set KMS_MASTER_KEY from docker secrets if provided
-docker_kms_master_encryption_env() {
-    KMS_MASTER_KEY_FILE="/run/secrets/$MINIO_KMS_MASTER_KEY_FILE"
-
-    if [ -f "$KMS_MASTER_KEY_FILE" ]; then
-        MINIO_KMS_MASTER_KEY="$(cat "$KMS_MASTER_KEY_FILE")"
-        export MINIO_KMS_MASTER_KEY
-    fi
-}
-
 # su-exec to requested user, if service cannot run exec will fail.
 docker_switch_user() {
     if [ ! -z "${MINIO_USERNAME}" ] && [ ! -z "${MINIO_GROUPNAME}" ]; then

--- a/internal/config/constants.go
+++ b/internal/config/constants.go
@@ -41,7 +41,6 @@ const (
 
 	EnvUpdate = "MINIO_UPDATE"
 
-	EnvKMSMasterKey  = "MINIO_KMS_MASTER_KEY" // legacy
 	EnvKMSSecretKey  = "MINIO_KMS_SECRET_KEY"
 	EnvKESEndpoint   = "MINIO_KMS_KES_ENDPOINT"
 	EnvKESKeyName    = "MINIO_KMS_KES_KEY_NAME"

